CMakeLists.txt

```diff
@@ -7,6 +7,7 @@ set( OPT_LEVEL "3" CACHE STRING "Compiler optmisation level." )
 option( SANITIZERS "Enable UBSAN" OFF )
 option( LTO "Enable link time optimisation." OFF )
 option( CCACHE "Enable ccache." ON )
+option( SEMIHOSTING "Enable semihosting." ON )
```

Comment:
If this is going to be set automatically per board (which is fine) it should be done like STACK_SIZE etc. Using set.
(option makes it seem like you can do cmake ... -DSEMIHOSTING=OFF, but you can't)

Also when you do that, you can remove the ON values for the current boards, just set it once, and over in your rpi fork just set OFF for that one.